### PR TITLE
Check Trumbowyg Settings

### LINKED ISSUE #5262: HtmlBody update part is broken
Version `1.0.0-rc1-11459`

part-1

part-2

This error keeps happening regardless what choice I used 

part-3

### PR DESCRIPTION
**Pull request by agriffard**
Fixes #5262

Check options are not empty before to parse the json in Trumbowyg editor settings.

### CODE REVIEW

**Comment by agriffard** [comment]
One difficulty is that `partFieldDefinition.Editor()` will return "Trumbowyg" only if it has previously been updated to it.
So if you change from Standard to Trumbowyg, it will return Standard and not call the update and if you change for Trumbowyg to Standard, it will return Trumbowyg and call the update.

**Comment by deanmarcussen** [comment]
Makes sense, so we should always run the settings update, regardless of which editor has been (previously) selected.

So should we change the `HtmlField` to also do that? and the null check you have added here?

**Comment by ns8482e** [comment]
Not sure if it's related But there is also an open issue #4893 when Trumbowyg and Wysiwyg on same page. It would be nice just to keep only one editor option that use Trumbowyg.js to avoid future issues

**Comment by deanmarcussen** [comment]
> One difficulty is that `partFieldDefinition.Editor()` will return "Trumbowyg" only if it has previously been updated to it.

I looked at this to understand the difference between the field drivers and the part drivers, where the editor value is updated on a field driver, but is not on a part driver.

There is a line missing in the `ContentParts.AdminController` which will fix that issue. 

@agriffard you mind if I push something to this pr to fix that?

**Comment by agriffard** [comment]
@deanmarcussen Yes, you can.

**Comment by deanmarcussen** [comment]
@agriffard Updated to fix the admin controller so it gives the current (changed) value of the editor type, as we do for fields

File: src/OrchardCore.Modules/OrchardCore.ContentFields/Settings/HtmlFieldTrumbowygEditorSettingsDriver.cs [hunk 1]
```diff
@@ -38,11 +38,14 @@ public override async Task<IDisplayResult> UpdateAsync(ContentPartFieldDefinitio
...
-                
+
                 try
                 {
-                    settings.Options = model.Options;
-                    JObject.Parse(settings.Options);
+                    if (!string.IsNullOrEmpty(model.Options))
+                    {
+                        settings.Options = model.Options;
+                        JObject.Parse(settings.Options);
```
Comment (Jetski5822) [inline src/OrchardCore.Modules/OrchardCore.ContentFields/Settings/HtmlFieldTrumbowygEditorSettingsDriver.cs]: Is this needed?

Comment (deanmarcussen) [reply src/OrchardCore.Modules/OrchardCore.ContentFields/Settings/HtmlFieldTrumbowygEditorSettingsDriver.cs]: No, no longer needed after fixing the admin controller. I've removed it

**Comment by deanmarcussen** [comment]
I think this is good to go if you agree with the changes I made @agriffard

**Pull request merged**